### PR TITLE
sort style classes

### PR DESCRIPTION
**Pull request by alexlamsl**
So rather than going with a simplistic global frequency count, this optimises for longest chain output.

Similar to #594, we get some savings in compressed outputs in most cases. What surprised me was the Google benchmark :open_mouth:

Usual warning about my choice and taste of variable names - suggestions welcome :sweat: 

TODOs:
- [x] apply `chain.js` to `sortAttributes`
- [x] combine first-pass scanning of `sortAttributes` and `sortClassName`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,9 +70,14 @@ How does HTMLMinifier compare to other solutions — [HTML Minifier from Will Pe
...
 | `removeTagWhitespace`          | Remove space between attributes whenever possible. **Note that this will result in invalid HTML!** | `false` |
-| `sortAttributes`               | Sort attributes by frequency | `false` |
+| `sortAttributes`               | [Sort attributes by frequency](#sorting-attributes--style-classes) | `false` |
+| `sortClassName`                | [Sort style classes by frequency](#sorting-attributes--style-classes) | `false` |
 | `useShortDoctype`              | [Replaces the `doctype` with the short (HTML5) doctype](http://perfectionkills.com/experimenting-with-html-minifier/#use_short_doctype) | `false` |
 
+### Sorting attributes / style classes
+
+Minifier options like `sortAttributes` and `sortClassName` won't impact the plain-text size of the output. However, they form long repetitive chains of characters that should improve compression ratio of gzip used in HTTP compression.
+
```
Comment (alexlamsl) [inline README.md:80]: @kangax as per https://github.com/kangax/html-minifier/commit/139d31496192863f92ba59da38cd608b328af525#commitcomment-16956122

**Comment by kangax** [comment]
Looks great. I would just rename `Chain` to `TokenChain` for clarity.

**Comment by alexlamsl** [comment]
@kangax the rebase was fun (since the file was introduced in the very first commit), but it's all done :sunglasses:

**Comment by kangax** [comment]
:+1:

**Pull request merged**